quiz.rb

```diff
@@ -1,20 +1,64 @@
 # Question 1
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
+class Dog
+  def initialize (name, breed, age)
+    @name = name
+    @breed = breed
+    @age = age
+  end
+
+  attr_reader :name
+  attr_reader :breed
+  attr_reader :age
+
+end
+
 
 # Question 2
 # Create a new Dog object.
 
+lab = Dog.new("Disco", "labrador", 3)
+
+
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
+class Person
+  def initialize(name, age, location)
+    attr_writer :name
+    attr_writer :location
+    attr_reader :age
+  end
+
+  def set_location(location)
+    @location = location
+  end
+
+end
+
```

Comment:
X
`attr_reader`, `attr_accessor`, etc belong _outside_ the `initialize` method.
Also, since we're using `attr` to create an accessor method for location, there's no need to create a `set_location` method.